interface/client/lib/helpers/helperFunctions.js

```diff
@@ -50,6 +50,21 @@ Helpers.formatUrl = function(url){
 };
 
 /**
+Sanatizes URLs to prevent phishing and XSS attacks
+
+@method sanatizeUrl
+@param {String} url
+**/
+Helpers.sanatizeUrl = function(url){    
+    // add http:// if no protocol is present
+    if(_.isString(url)) {
+        url = url.replace(/^(javascript:)?(data:)?\/\//i, 'http:');
+    }
+
+    return url;
```

Comment:
I will look into a proper JS library. Currently the URL is never a location object, but i can add checks for them too.

Whitelisting would make sense if we fully know what people will use and currently we don't want to restrict unknown URIs. E.g. ftp, etc.
As it looks only JavaScript is semi-dangerous as it can spawn ui actions, but this js has also not added privileges than the webpage itself.
So removing JavaScript seems to be the most dangerous one, can you think of any other?